llvm/include/llvm/Target/TargetPfmCounters.td

```diff
@@ -28,6 +28,22 @@ class PfmIssueCounter<string resource_name, string counter>
   string ResourceName = resource_name;
 }
 
+class ValidationEvent <int event_number> {
```

Comment:
This needs doc to explain what a `ValidationEvent` and `PfmValidationCounter` are.

Reply:
Added additional comments on what these are for. Please let me know if they don't clarify things enough.